homeassistant/components/sensor/nmbs.py

```diff
@@ -0,0 +1,218 @@
+"""
+Get ride details and liveboard details for NMBS (Belgian railway).
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nmbs/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'NMBS'
+DEFAULT_NAME_LIVE = "NMBS Live"
+
+DEFAULT_ICON = "mdi:train"
+DEFAULT_ICON_ALERT = "mdi:alert-octagon"
+
+CONF_STATION_FROM = 'station_from'
+CONF_STATION_TO = 'station_to'
+CONF_STATION_LIVE = 'station_live'
+
+REQUIREMENTS = ["pyrail==0.0.3"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_FROM): cv.string,
+    vol.Required(CONF_STATION_TO): cv.string,
+    vol.Optional(CONF_STATION_LIVE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def get_time_until(departure_time=None):
+    """Calculate the time between now and a train's departure time."""
+    if departure_time is None:
+        return 0
+
+    delta = dt_util.utc_from_timestamp(int(departure_time)) - dt_util.now()
+    return round((delta.total_seconds() / 60))
+
+
+def get_delay_in_minutes(delay=0):
+    """Get the delay in minutes from a delay in seconds."""
+    return round((int(delay) / 60))
+
+
+def get_ride_duration(departure_time, arrival_time, delay=0):
+    """Calculate the total travel time in minutes."""
+    duration = dt_util.utc_from_timestamp(
+        int(arrival_time)) - dt_util.utc_from_timestamp(int(departure_time))
+    duration_time = int(round((duration.total_seconds() / 60)))
+    return duration_time + get_delay_in_minutes(delay)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the NMBS sensor with iRail API."""
+    from pyrail import iRail
+    api_client = iRail()
+
+    name = config[CONF_NAME]
+    station_from = config[CONF_STATION_FROM]
+    station_to = config[CONF_STATION_TO]
+    station_live = config[CONF_STATION_LIVE]
```

Comment:
Am I right to think this one should still use `.get`?
It has no default value and we need it to be `None` if it's not defined (it's optional).

```suggestion
    station_live = config.get(CONF_STATION_LIVE)
```

Reply:
Yes, you're right. Here we should use `dict.get`. :+1: